manim/mobject/mobject.py

```diff
@@ -383,15 +383,15 @@ def add_to_back(self, *mobjects: "Mobject") -> "Mobject":
         :meth:`add`
 
         """
+        if self in mobjects:
+            raise ValueError("A mobject shouldn't contain itself")
+
         for mobject in mobjects:
-            if self in mobjects:
-                raise ValueError("Mobject cannot contain self")
             if not isinstance(mobject, Mobject):
                 raise TypeError("All submobjects must be of type Mobject")
 
-        filtered = list_update(mobjects, self.submobjects)
         self.remove(*mobjects)
-        self.submobjects = list(filtered) + self.submobjects
+        self.submobjects = list(dict.fromkeys(mobjects)) + self.submobjects
```

Comment:
Can you leave a comment explaining the reasoning of this line ? (deleting duplicates while keeping the order)